mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -109,18 +110,42 @@ const getNewParams = (prevParams: SnowflakeImportParams, database?: string | nul
 */
 const SnowflakeImportTaskpane = (props: SnowflakeImportTaskpaneProps): JSX.Element => {
 
-    const {params, setParams: setParamsWithoutRefreshOptionsAndDefaults, edit} = useSendEditOnClick<SnowflakeImportParams, undefined>(
+    const {params, setParams: setParamsWithoutRefreshOptionsAndDefaults, edit, loading: executingQuery, error} = useSendEditOnClick<SnowflakeImportParams, undefined>(
         () => getDefaultParams(),
         StepType.SnowflakeImport, 
         props.mitoAPI,
         props.analysisData,
+        undefined,
+        (newParamsOrFunction) => {
+            // When an Undo or Redo occurs, after getting the new params, use them 
+            // to load and set the available options and defaults
+            let newParams: SnowflakeImportParams | undefined = undefined
+            if (typeof newParamsOrFunction === 'function') {
+                newParams = newParamsOrFunction()
+            } else {
+                newParams = newParamsOrFunction
+            }
+            if (newParams !== undefined) {
+                void loadAndSetOptionsAndDefaults(newParams)
+            }
```

Comment:
This is an example of really bad API design (that you inherited!) as a function of types that I picked when I designed this. The consumer has to do a bunch of work to check if the new params are a type of function -- but in practice (if you actually look at what is returned), they never will be. 

If you just:
1. Change the type of `onUndoAndRedo?: (params: typeof defaultParams) => void` to have `params: ParamType`
2. Change `await mitoAPI.getParams<typeof defaultParams>(stepType, stepID, {});` to take `ParamType` as well.

Then, you don't need to do this checking. Which is... much more plesant!